tool/tctl/common/device/resource.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package device
+
+import (
+	"time"
+
+	"github.com/gravitational/trace"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	devicepb "github.com/gravitational/teleport/api/gen/proto/go/teleport/devicetrust/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+// Resource is a wrapper around devicepb.Device that implements types.Resource.
+type Resource struct {
+	// ResourceHeader is embedded to implement types.Resource
+	types.ResourceHeader
+	// Spec is the device specification
+	Spec ResourceSpec `json:"spec"`
+}
+
+// ResourceSpec is the device resource specification.
+type ResourceSpec struct {
+	// OsType is the device's operating system type.
+	OsType string `json:"os_type"`
+	// AssetTag is the device's asset tag.
```

Comment:
Sure. Added them because I've gotten nixed before for not having them, believing always having them was convention. They're also pretty standard in our codebase but I do feel they are redundant.